### PR TITLE
(maint) Update PSON doc

### PR DESCRIPTION
**Pull request by joshcooper**
Describe how pson forked from json and was modified to allow arbitrary
binary content. Make it clearer how pson differs from json.

### CODE REVIEW

**Comment by joshcooper** [comment]
@GabrielNagy could you review?

File: api/docs/pson.md [hunk 1]
```diff
@@ -30,11 +39,47 @@ characters:
...
+    #=> #<Encoding:UTF-8>
+
+Finally, PSON does not roundtrip values that are not arrays or hashes in the same way as JSON:
+
+    JSON.parse(JSON.generate("\x1E"))
+    #=> "\u001E"
+    PSON.parse(PSON.generate("\x1E"))
+    #=> PSON::ParserError (source '"\u001e"' not in PSON!)
+
+whereas PSON can roundtriip an array (or hash) containing that value:
```
Comment (GabrielNagy) [inline api/docs/pson.md:79]: nit: roundtrip

**Review by GabrielNagy — APPROVED** [review]
Found a typo, looks good otherwise :+1:

File: api/docs/pson.md (hunk 1 above)
Comment (joshcooper) [reply api/docs/pson.md]: Thanks! updated

**Pull request merged**